### PR TITLE
Issue 29 add type script support

### LINKED ISSUE #29: ✨ [Feature request]: Add TypeScript Support
Currently the `yarn create-play` command creates the play in JavaScript(JSX). Please add the support to create play scaffolding based on TypeScript.

### PR DESCRIPTION
**Pull request by koustov**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
Closes #29 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
Please check the error log @koustov 

[14:12:32] Cloning github.com/atapas/react-play (Branch: ISSUE-29-Add-TypeScript-support, Commit: f4aabde)
[14:12:32] Cloning completed: 568.72ms
[14:12:33] Installing build runtime...
[14:12:34] Build runtime installed: 1.689s
[14:12:35] Looking up build cache...
[14:12:36] Build Cache not found
[14:12:36] Installing dependencies...
[14:12:36] yarn install v1.22.17
[14:12:36] [1/4] Resolving packages...
[14:12:37] [2/4] Fetching packages...
[14:12:55] [3/4] Linking dependencies...
[14:12:55] warning " > @testing-library/user-event@13.5.0" has unmet peer dependency "@testing-library/dom@>=7.21.4".
[14:12:55] warning "react-helmet > react-side-effect@2.1.1" has incorrect peer dependency "react@^16.3.0 || ^17.0.0".
[14:12:55] warning "react-organizational-chart > @emotion/css > @emotion/babel-plugin@11.7.2" has unmet peer dependency "@babel/core@^7.0.0".
[14:12:55] warning "react-organizational-chart > @emotion/css > @emotion/babel-plugin > @babel/plugin-syntax-jsx@7.16.7" has unmet peer dependency "@babel/core@^7.0.0-0".
[14:12:55] warning "react-scripts > tailwindcss@3.0.12" has unmet peer dependency "autoprefixer@^10.0.2".
[14:12:55] warning "react-scripts > eslint-config-react-app > eslint-plugin-flowtype@8.0.3" has unmet peer dependency "@babel/plugin-syntax-flow@^7.14.5".
[14:12:55] warning "react-scripts > eslint-config-react-app > eslint-plugin-flowtype@8.0.3" has unmet peer dependency "@babel/plugin-transform-react-jsx@^7.14.9".
[14:12:55] warning " > react-shimmer-effect@1.0.9" has incorrect peer dependency "react@^16.8.1".
[14:12:55] warning " > react-shimmer-effect@1.0.9" has incorrect peer dependency "react-dom@^16.8.1".
[14:13:11] [4/4] Building fresh packages...
[14:13:16] $ ./node_modules/.bin/gulp build
[14:13:16] [08:43:16] Using gulpfile ~/path0/gulpfile.js
[14:13:16] [08:43:16] Task never defined: build
[14:13:16] [08:43:16] To list available tasks, try running: gulp --tasks
[14:13:16] error Command failed with exit code 1.
[14:13:16] info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
[14:13:16] Error: Command "yarn install" exited with 1

File: README.md [hunk 1]
```diff
@@ -83,6 +83,14 @@ Or
 ```bash
 yarn install
```
Comment (atapas) [inline README.md:84]: Add yarn equivalent command too

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+var gulp = require('gulp');
```
Comment (atapas) [inline gulpfile.js:1]: Why gulp is needed?

File: package.json [hunk 3]
```diff
@@ -23,7 +26,8 @@
     "postbuild": "react-snap",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "create-play": "plop"
+    "create-play": "plop",
```
Comment (atapas) [inline package.json:29]: Do we need this post install script

File: package.json [hunk 4]
```diff
@@ -50,6 +54,7 @@
     ]
   },
   "devDependencies": {
+    "gulp": "^4.0.2",
```
Comment (atapas) [inline package.json:57]: Do we need gulp?

**Review by atapas — CHANGES_REQUESTED** [review]
A few questions and additionally do not commit yarn lock

**Comment by koustov** [comment]
having an issue with this PR. Will recreate one more

**Pull request closed without merging**